ESACCI/read-biomass-ESACCI.py

```diff
@@ -0,0 +1,255 @@
+        dx = R*math.cos(0.01745*lat1[ny])*dlon
+        areas = dx*dy*1.0e-4
+
+        #+++++ to generate data in 0.5x0.5 ++++
+        iy   = int(lat1[ny]*2+90.*2)
+        ix   = int(lon1[nx]*2+180.*2)
+
+        # obtain total biomass mass at each grid cell
+        biomass[iy,ix] = biomass[iy,ix] + agb[ny,nx]*areas
+
+#biomass = rebin(abgb, (int(nlat/2), int(nlon/2)))
+
+print('Total biomass')
+print(np.sum(biomass*1.0e-9))
+
+# convert total mass (Mg) to density (Mg/m2)
+biomass[...] = biomass[...]/area[...]
+
+# convert biomass to biomass carbon 
+biomass[:,:] = biomass[:,:]/2.
+
+# convert the unit from Mg/m2 to Kg/m2
+biomass[:,:] = biomass[:,:]*1000.
+
+print(biomass.shape)
+
+#data[:,:] = rebin(agb, (nlat, nlon)) 
+
+data[0,:,:] = biomass[:,:]
+
+print('Total biomass carbon')
+print(np.sum(biomass*area*1.0e-12))
+
+with Dataset(DataDir + "/biomass.nc", mode="w") as dset:
+
+    # Create netCDF dimensions
+    dset.createDimension("time",size=  t.size)
+    dset.createDimension("lat" ,size=lat.size)
+    dset.createDimension("lon" ,size=lon.size)
+    dset.createDimension("nb"  ,size=2       )
+
+    # Create netCDF variables
+    T  = dset.createVariable("time"       ,t.dtype   ,("time"     ))
+    TB = dset.createVariable("time_bounds",t.dtype   ,("time","nb"))
+    X  = dset.createVariable("lat"        ,lat.dtype ,("lat"      ))
+    XB = dset.createVariable("lat_bounds" ,lat.dtype ,("lat","nb" ))
+    Y  = dset.createVariable("lon"        ,lon.dtype ,("lon"      ))
+    YB = dset.createVariable("lon_bounds" ,lon.dtype ,("lon","nb" ))
+    D  = dset.createVariable("biomass"    ,data.dtype,("time","lat","lon"), fill_value = -999.)
+
+    # Load data and encode attributes
+    # time
+    T [...]         = t
+    T.units         = "days since 1850-01-01"
+    T.calendar      = "noleap"
+    T.bounds        = "time_bounds"
+    TB[...]         = tbnd
+    T.standard_name = "time"
+    T.long_name     = "time"
+
+    # lat
+    X [...]    = lat
+    X.units    = "degrees_north"
+    XB[...]    = latbnd
+    X.standard_name = "latitude"
+    X.long_name     = "latitude"
+
+    # lon
+    Y [...]    = lon
+    Y.units    = "degrees_east"
+    YB[...]    = lonbnd
+    Y.standard_name = "longitude"
+    Y.long_name     = "longitude"
+
+    # data
+    D[...] = data
+    D.units = "kg m-2"
+    D.standard_name = "global forest biomass carbon"
+    D.long_name     = long_name
+    D.actual_range = np.asarray([data.min(),data.max()])
+    
+    dset.title       = "ESA CCI above-ground biomass product level 4, year 2017"
+    dset.version     = "1"
+    dset.institutions = "%s; %s" % (instit1, instit2)
+    dset.source      = "This dataset contains a global map of above-ground biomass of the epoch 2017 obtained from L-and C-band spaceborne SAR backscatter, placed onto a regular grid."
+    dset.history     = """
+%s: downloaded source from %s;
+%s: convert original tif files to netcdf data using gdal_translate;
```

Comment:
Is this true? this script does not call gdal_translate but I guess you might have done that as a previous step. If true, then fine. If false, then remove.